app/src/main/java/wannabit/io/cosmostaion/chain/ChainFactory.java

```diff
@@ -0,0 +1,276 @@
+            return new Injective();
+        } else if (chainInfo.startsWith("juno-") || chainInfo.startsWith("juno1") || chainInfo.equalsIgnoreCase(TOKEN_JUNO)) {
+            return new Juno();
+        } else if (chainInfo.startsWith("kava-") || chainInfo.startsWith("kava1") || chainInfo.equalsIgnoreCase(TOKEN_KAVA)) {
+            return new Kava();
+        } else if (chainInfo.startsWith("kichain-") || chainInfo.startsWith("ki1") || chainInfo.equalsIgnoreCase(TOKEN_KI)) {
+            return new Ki();
+        } else if (chainInfo.startsWith("darchub") || chainInfo.startsWith("darc1") || chainInfo.equalsIgnoreCase(TOKEN_DARC)) {
+            return new Konstellation();
+        } else if (chainInfo.startsWith("lum-network-") || chainInfo.startsWith("lum1") || chainInfo.equalsIgnoreCase(TOKEN_LUM)) {
+            return new Lum();
+        } else if (chainInfo.startsWith("panacea-") || chainInfo.startsWith("panacea1") || chainInfo.equalsIgnoreCase(TOKEN_MEDI)) {
+            return new Medibloc();
+        } else if (chainInfo.startsWith("omniflixhub-") || chainInfo.startsWith("omniflix1") || chainInfo.equalsIgnoreCase(TOKEN_FLIX)) {
+            return new Omniflix();
+        } else if (chainInfo.startsWith("osmosis-") || chainInfo.startsWith("osmo1") || chainInfo.equalsIgnoreCase(TOKEN_OSMOSIS)) {
+            return new Osmosis();
+        } else if (chainInfo.startsWith("core-") || chainInfo.startsWith("persistence1") || chainInfo.equalsIgnoreCase(TOKEN_XPRT)) {
+            return new Persistence();
+        } else if (chainInfo.startsWith("pio-mainnet-") || chainInfo.startsWith("pb1") || chainInfo.equalsIgnoreCase(TOKEN_HASH)) {
+            return new Provenance();
+        } else if (chainInfo.startsWith("regen-") || chainInfo.startsWith("regen1") || chainInfo.equalsIgnoreCase(TOKEN_REGEN)) {
+            return new Regen();
+        } else if (chainInfo.startsWith("titan-") || chainInfo.startsWith("rizon1") || chainInfo.equalsIgnoreCase(TOKEN_RIZON)) {
+            return new Rizon();
+        } else if (chainInfo.startsWith("secret-") || chainInfo.startsWith("secret1") || chainInfo.equalsIgnoreCase(TOKEN_SECRET)) {
+            return new Secret();
+        } else if (chainInfo.startsWith("sifchain-") || chainInfo.startsWith("sif1") || chainInfo.equalsIgnoreCase(TOKEN_SIF)) {
+            return new Sif();
+        } else if (chainInfo.startsWith("sentinelhub-") || chainInfo.startsWith("sent1") || chainInfo.equalsIgnoreCase(TOKEN_DVPN)) {
+            return new Sentinel();
+        } else if (chainInfo.startsWith("stargaze-") || chainInfo.startsWith("stars1") || chainInfo.equalsIgnoreCase(TOKEN_STARGAZE)) {
+            return new Stargaze();
+        } else if (chainInfo.startsWith("iov-") || chainInfo.startsWith("star1") || chainInfo.equalsIgnoreCase(TOKEN_IOV)) {
+            return new Starname();
+        } else if (chainInfo.startsWith("umee-") || chainInfo.startsWith("umee1") || chainInfo.equalsIgnoreCase(TOKEN_UMEE)) {
```

Comment:
If Chain classes have chainIds and Chain tokens, can we find chain from all chains by chain.chainIds.contains(chainInfo) || chain.token.equals(chainInfo)